jax/_src/numpy/lax_numpy.py

```diff
@@ -3207,6 +3207,41 @@ def meshgrid(*args, **kwargs):
   return output
 
 
+class _Mgrid:
+  """Return dense multi-dimensional "meshgrid".
+
+  LAX-backend implementation of `mgrid()`."""
+
+  def __getitem__(self, key):
+    if isinstance(key, slice):
+      start = key.start or 0
+      stop = key.stop
+      step = key.step or 1
+      if np.iscomplex(step):
+        return linspace(start, stop, int(abs(step)))
+      return arange(start, stop, step)
+
+    key = tuple(
+        core.concrete_or_error(None, k, "key of jnp.mgrid") for k in key)
```

Comment:
`core.concrete_or_error` does not recurse into the object, so you'll need to call it on ``k.start``, ``k.stop``, and ``k.step`` individually.

Reply:
Updated!